### PR TITLE
fix: add custom specPattern if user opt out renaming spec ext

### PR DESCRIPTION
**Pull request by estrada9166**
- Closes  [UNIFY-1679](https://cypress-io.atlassian.net/jira/software/projects/UNIFY/boards/20?assignee=5ea883deddd45f0b94f5d98d&selectedIssue=UNIFY-1679)

### PR Tasks

- [ ] Have tests been added/updated?
- [ ] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [ ] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [ ] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
- [ ] Have new configuration options been added to the [`cypress.schema.json`](https://github.com/cypress-io/cypress/blob/develop/cli/schema/cypress.schema.json)?

### CODE REVIEW

**Review by tbiethman — APPROVED** [review]
Ran through opt-out workflow, results are consistent the reqs on the JIRA and the added system tests :+1:

File: packages/data-context/src/actions/MigrationActions.ts [hunk 1]
```diff
@@ -247,6 +250,13 @@ export class MigrationActions {
     const from = path.join(projectRoot, 'cypress', 'integration')
     const to = path.join(projectRoot, 'cypress', 'e2e')
 
+    this.ctx.update((coreData) => {
+      coreData.migration.flags = {
+        ...coreData.migration.flags,
+        shouldAddCustomE2eSpecPattern: true,
```
Comment (lmiller1990) [inline packages/data-context/src/actions/MigrationActions.ts:256]: Nit: we usually write this as `E2E`, not `E2e`, in this code base from what I can see.

File: packages/data-context/src/sources/migration/codegen.ts [hunk 2]
```diff
@@ -444,8 +444,8 @@ export function reduceConfig (cfg: LegacyCypressConfigJson): ConfigOptions {
   }, { global: {}, e2e: {}, component: {} })
 }
 
-export function getSpecPattern (cfg: LegacyCypressConfigJson, testType: TestingType) {
-  const specPattern = cfg[testType]?.testFiles ?? cfg.testFiles ?? (cfg.projectId ? '**/*.{js,ts,tsx,jsx}' : '**/*.cy.{js,jsx,ts,tsx}')
+export function getSpecPattern (cfg: LegacyCypressConfigJson, testType: TestingType, shouldAddCustomE2eSpecPattern?: boolean) {
```
Comment (lmiller1990) [inline packages/data-context/src/sources/migration/codegen.ts:447]: Another nit: we generally call this `testingType`, not `testType` 😅

Comment (lmiller1990) [inline packages/data-context/src/sources/migration/codegen.ts:447]: Should we have a default value for `shouldAddCustomE2eSpecPattern`? eg 

```js
export function getSpecPattern (cfg: LegacyCypressConfigJson, testType: TestingType, shouldAddCustomE2eSpecPattern?: boolean = false)
```

Where the default is a bit defensive, so I guess it'd be `false` here?

**Review by lmiller1990 — APPROVED** [review]
Looks good, I have 2 nitpick comments (not a big deal) and one potential improvement with a default parameter - what do you think? Not strictly necessary, so I'll approve it for now.

**Pull request merged**